swap/peer.go

```diff
@@ -49,6 +51,7 @@ func NewPeer(p *protocols.Peer, s *Swap, beneficiary common.Address, contractAdd
 		swap:            s,
 		beneficiary:     beneficiary,
 		contractAddress: contractAddress,
+		logger:          newPeerLogger(s, p.ID()),
```

Comment:
there's an issue here: the result of `p.ID()` is not structurally identical to the id that regular SWAP logger is using (which is `Swap.owner.Address`).

not sure which one is more appropriate or if we can even get the same kind of id for both the node and the peer.

Reply:
`beneficiary` should be structurally identical to `Swap.owner.Address`, as the `beneficiary` is the owner of the peer's chequebook (see `protocol.go:119-124`).

I am also not sure which is more appropriate to use and if we can get something similar as peer.ID() of the node.

Reply:
see @acud's comment down below.

Reply:
> We should use `ID()` here, because in all of the Swarm logs, we deal with the enode ID, not the owner address of the contract, which is only relevant for swap related logs.
> 
> But if we want to track who is exchanging with who, we should use `p.ID()`.

i'm not sure i understand your comment.

do you mean the same thing by `ID()` and `p.ID()`?

Reply:
yes, now is ok until we harmonize the address space in the loggers between enodes and bzz address overall.